src/App.tsx

```diff
@@ -1,16 +1,27 @@
 import './App.scss';
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
+import { Movie } from './types/Movie';
 import moviesFromServer from './api/movies.json';
 
 export const App = () => {
+  const [movies, setMovie] = useState(moviesFromServer);
+
+  const handleMovie = (movie: Movie) => {
+    setMovie([
+      ...movies,
+      movie,
+    ]);
```

Comment:
If the new state is computed using the previous state, you can pass a function to setMovie
```suggestion
    setMovie((currentMovies) => [
      ...currentMovies,
      movie,
    ]);
```